Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,238 @@
+// Created by Dan Federman on 2/2/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+public enum TypeDescription: Codable, Equatable, CustomStringConvertible {
+  case simple(name: String)
+  indirect case member(name: String, baseType: TypeDescription)
+  indirect case composition([TypeDescription])
+  indirect case optional(TypeDescription)
+  indirect case implicitlyUnwrapped(TypeDescription)
+  indirect case array(TypeDescription)
+  indirect case dictionary(key: TypeDescription, value: TypeDescription)
+  indirect case tuple([TypeDescription])
+  case unknown(text: String)
+
+  /// A description of this type that can be used for code-generation to represent this type.
+  public var description: String {
```

Comment:
I've trained myself to avoid using `description` for core API use cases under the assumption that it's not guaranteed to be stable ([example](https://twitter.com/mattt/status/1174041151214538752?s=20)). What do you think about having a specific property like `text`, even if `description` has the same value? Then we can choose to evolve `description` if we want and not break use cases.

Reply:
I'm following `SwiftSyntax`'s lead here. Nodes all have a `description` that is source-accurate. While counter-intuitive, I'm trying to make this layer match the intent.

I think in the future if we wanted to add more info we'd need to evolve `debugDescription` rather than `description`, which seems okay? Open to changing here but wanted to explain my thinking before making the change blindly.

Reply:
Alright let's see what y'all think of this naming idea: 167f0172d5e21c5d0b1b7bb9cd49f4ceb5d4716e